src/runtime/test/data-layer-test.ts

```diff
@@ -37,6 +37,8 @@ describe('entity', async () => {
     const clone = list[0];
     assert.isDefined(clone);
     assert.deepEqual(clone.rawData, {value: 'hello world'});
-    assert.notEqual(entity, clone);
+
```

Comment:
This does not appear to be a valid test since the two objects are not comparable...

Reply:
Once we provide the full storage stack to the shell then we should be able to use a similar stack here, at which point the types will be the same. Not sure of the value overall of checking they're not deeply equal though - probably just looking at the ids would be sufficient.

Reply:
The IDs are the same.  I'll add a TODO to this for future investigation.